elyra/pipeline/handlers.py

```diff
@@ -138,4 +174,33 @@ async def get(self, processor, component_id):
 
         self.set_status(200)
         self.set_header("Content-Type", 'application/json')
+
         self.finish(properties_json)
+
+
+class PipelineValidationHandler(HttpErrorMixin, APIHandler):
+    """Handler to expose method calls to execute pipelines as batch jobs"""
+
+    @web.authenticated
+    async def get(self):
+        msg_json = dict(title="GET requests are not supported.")
+        self.write(msg_json)
+        self.flush()
+
+    @web.authenticated
+    async def post(self):
+        self.log.debug("Pipeline Validation Handler now executing post request")
+
+        root_dir = self.settings['server_root_dir']
+        expanded_path = get_expanded_path(root_dir)
+
+        pipeline_definition = self.get_json_body()
+        self.log.debug("Pipeline payload: %s", pipeline_definition)
+
+        response = await PipelineValidationManager.instance().validate(pipeline=pipeline_definition,
+                                                                       root_dir=expanded_path)
```

Comment:
After rebasing to master and instantiating the singleton instance in the application startup sequence, you _could_ obtain `root_dir` as a member variable of the singleton where it would be accessible via `self.root_dir` to the validation manager's method.  This might eliminate some potential confusion since other singletons that need the root directory are doing similarly.

Reply:
mimic'ed the other singletons in the startup sequence